ckanext/who_afro/plugin.py

```diff
@@ -92,47 +92,51 @@ def before_resource_update(self, context, current, resource):
     # IActions
     def get_actions(self):
         return {
-            'user_list': who_afro_actions.user_list,
-            'dataset_duplicate': who_afro_actions.dataset_duplicate,
-            'package_create': who_afro_actions.package_create,
-            'dataset_tag_replace': who_afro_actions.dataset_tag_replace,
-            'user_show_me': who_afro_actions.user_show_me,
+            "user_list": who_afro_actions.user_list,
+            "dataset_duplicate": who_afro_actions.dataset_duplicate,
+            "package_create": who_afro_actions.package_create,
+            "dataset_tag_replace": who_afro_actions.dataset_tag_replace,
+            "user_show_me": who_afro_actions.user_show_me,
         }
 
     # IValidators
     def get_validators(self):
         return {
-            'autogenerate_name_from_title': who_afro_validators.autogenerate_name_from_title,
-            'autofill': who_afro_validators.autofill,
-            'autogenerate': who_afro_validators.autogenerate,
-            'isomonth': who_afro_validators.isomonth
+            "autogenerate_name_from_title": who_afro_validators.autogenerate_name_from_title,
+            "autofill": who_afro_validators.autofill,
+            "autogenerate": who_afro_validators.autogenerate,
+            "isomonth": who_afro_validators.isomonth,
         }
 
     # IPackageContoller
     def after_dataset_delete(self, context, data_dict):
-        package_data = toolkit.get_action('package_show')(context, data_dict)
-        if package_data.get('private'):
-            package_data['state'] = 'deleted'
-            context['package'].state = 'deleted'
+        package_data = toolkit.get_action("package_show")(context, data_dict)
+        if package_data.get("private"):
+            package_data["state"] = "deleted"
+            context["package"].state = "deleted"
             who_afro_upload.add_activity(context, package_data, "changed")
 
     def after_dataset_update(self, context, data_dict):
-        if data_dict.get('private'):
+        if data_dict.get("private"):
             who_afro_upload.add_activity(context, data_dict, "changed")
 
     def after_dataset_create(self, context, data_dict):
-        if data_dict.get('private'):
+        if data_dict.get("private"):
             who_afro_upload.add_activity(context, data_dict, "new")
 
     def before_dataset_index(self, data_dict):
-        data_dict['programme'] = json.loads(data_dict['programme'])
-        data_dict['country'] = json.loads(data_dict['country'])
+        if isinstance(data_dict["programme"], str):
```

Comment:
Maybe add a comment:
```suggestion
        # Load custom multivalued fields as objects before solr indexing. 
        if isinstance(data_dict["programme"], str):
```